src/lsptoolshost/roslynProtocol.ts

```diff
@@ -152,17 +152,24 @@ export interface NamedPipeInformation {
 
 export interface RestoreParams extends lsp.WorkDoneProgressParams, lsp.PartialResultParams {
     /**
-     * An optional file path to restore.
-     * If none is specified, the solution (or all loaded projects) are restored.
+     * The set of projects to restore.
+     * If none are specified, the solution (or all loaded projects) are restored.
      */
-    projectFilePath?: string;
+    projectFilePaths: string[];
 }
 
 export interface RestorePartialResult {
     stage: string;
     message: string;
 }
 
+export interface UnresolvedProjectDependenciesParams {
+    /**
+     * The set of projects that have unresolved dependencies and require a restore.
+     */
```

Comment:
Might still need a rename or update too since we're also handling other changes.